pages/search.tsx

```diff
@@ -56,67 +50,53 @@ function Index(query: {search: string | null}) {
             You can search by name or team name
           </p>
 
-          <form className="mt-10" onSubmit={search}>
+          <form className="mt-10" onSubmit={e => updateSearchValues(0, e)}>
             <input
               aria-label="Search for fpl player"
               value={searchValue}
               placeholder="Magnus Carlsen"
               onChange={(e) => {
                 setSearchValue(e.target.value);
               }}
-              onKeyDown={handleKeyDown}
               className="w-72 py-2 px-4 mr-4 text-fpl-purple border-2 border-fpl-purple rounded focus:outline-none"
             />
 
-            <Button onClick={search} shape="long" className="mt-4">
+            <Button onClick={e => updateSearchValues(0, e)} shape="long" className="mt-4">
               Search
             </Button>
           </form>
         </div>
         <div className="pb-24 px-8 text-center">
           <SearchState
             searchState={searchState}
-            searchPhrase={prevSearchState}
+            searchPhrase={submittedSearchValue}
+            page={pageValue}
           />
+          <Pagination searchState={searchState} searchPhrase={submittedSearchValue} currentPage={pageValue} updatePage={newPage => updateSearchValues(newPage)}/>
         </div>
       </div>
       <Footer />
     </div>
   );
 
-  function handleKeyDown(e: React.KeyboardEvent<HTMLInputElement>) {
-    if (e.key === "Enter") {
-      search(e);
-    }
-  }
-
-  function search(event?: any) {
+  function updateSearchValues(page: number, event?: any){
```

Comment:
Tror jeg ville splitta denne i to funksjoner som håndterer page number og søk separat. typ
```
submitNewSearchValue() {
    setSubmittedSearchValue(searchValue);
    setPageValue(0);
    updateQueryParam(searchValue, 0);
}

updatePageNumber(page: number) {
    setPageValue(page);
    updateQueryParam(searchValue, page);
}

```

Reply:
Enig! Done 👍